lib/collection/src/collection_manager/holders/segment_holder.rs

```diff
@@ -532,12 +567,12 @@ impl<'s> SegmentHolder {
     ///
     /// The `segment_data` function is called no more than once for each segment and its result is
     /// passed to `point_operation`.
-    pub fn apply_points<T, O, D>(
+    pub fn apply_points<T, D, O, OD>(
         &self,
         ids: &[PointIdType],
         mut segment_data: D,
         mut point_operation: O,
-        all_point_versions: bool,
+        mut point_delete_operation: OD,
```

Comment:
Why can't we just delete points without external function?

Reply:
I've removed the closure, it now deletes internally